pinot-core/src/main/java/org/apache/pinot/core/geospatial/transform/function/StAsGeoJsonFunction.java

```diff
@@ -0,0 +1,91 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.geospatial.transform.function;
+
+import com.google.common.base.Preconditions;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import org.apache.pinot.core.operator.ColumnContext;
+import org.apache.pinot.core.operator.blocks.ValueBlock;
+import org.apache.pinot.core.operator.transform.TransformResultMetadata;
+import org.apache.pinot.core.operator.transform.function.BaseTransformFunction;
+import org.apache.pinot.core.operator.transform.function.TransformFunction;
+import org.apache.pinot.core.plan.DocIdSetPlanNode;
+import org.apache.pinot.segment.local.utils.GeometrySerializer;
+import org.apache.pinot.segment.local.utils.GeometryUtils;
+import org.apache.pinot.spi.data.FieldSpec;
+import org.locationtech.jts.geom.Geometry;
+
+
+/**
+ * Returns the GEOJson representation of the geometry object.
+ */
+public class StAsGeoJsonFunction extends BaseTransformFunction {
+
+  public static final String FUNCTION_NAME = "ST_AsGeoJson";
+
+  private TransformFunction _transformFunction;
+  private String[] _results;
+
+  public String getName() {
+    return FUNCTION_NAME;
+  }
+
+  @Override
+  public void init(List<TransformFunction> arguments, Map<String, ColumnContext> columnContextMap) {
+    super.init(arguments, columnContextMap);
+    Preconditions.checkArgument(arguments.size() == 1,
+        "Exactly 1 argument is required for transform function: " + FUNCTION_NAME);
+
+    TransformFunction transformFunction = arguments.get(0);
+    Preconditions.checkArgument(transformFunction.getResultMetadata().isSingleValue(),
+        "Argument must be single-valued for transform function: " + FUNCTION_NAME);
+    Preconditions.checkArgument(transformFunction.getResultMetadata().getDataType() == FieldSpec.DataType.BYTES,
+        "The argument must be of bytes type");
+
+    _transformFunction = transformFunction;
+  }
+
+  @Override
+  public TransformResultMetadata getResultMetadata() {
+    return STRING_SV_NO_DICTIONARY_METADATA;
+  }
+
+  public String[] transformToStringValuesSV(ValueBlock valueBlock) {
+    if (_results == null) {
+      _results = new String[DocIdSetPlanNode.MAX_DOC_PER_CALL];
+    }
+    byte[][] values = _transformFunction.transformToBytesValuesSV(valueBlock);
+    // use single buffer instead of allocating separate StringBuffer per row
+    StringBuilderWriter buffer = new StringBuilderWriter();
+
+    try {
+      for (int i = 0; i < valueBlock.getNumDocs(); i++) {
+        Geometry geometry = GeometrySerializer.deserialize(values[i]);
+        GeometryUtils.GEO_JSON_WRITER.write(geometry, buffer);
+        _results[i] = buffer.getString();
+        buffer.clear();
+      }
+    } catch (IOException ioe) {
+      // should never happen
```

Comment:
Even though it should not happen, let's wrap it as a runtime exception and throw it to guarantee it works as expected. Same for other places